rspace-bench/src/main/scala/coop/rchain/rspace/bench/EvalBench.scala

```diff
@@ -0,0 +1,119 @@
+//rspaceBench/jmh:run EvalBench.replayMvcEppST -i 1 -wi 0 -f 0 -t 1
+//on hyper-threaded machines for standalone check use -f <number of physical cores>
+//for example
+//java -jar target/scala-2.12/rspacebench_2.12-0.1.0-SNAPSHOT.jar EvalBench -i 10 -wi 5 -f 2 -t 2
+class EvalBench {
+  import EvalBench._
+
+  def processErrors(errors: Vector[Throwable]): Unit = if (errors.nonEmpty) {
+    throw new RuntimeException(
+      errors
+        .map(_.toString())
+        .mkString("Errors received during evaluation:\n", "\n", "\n"))
+  }
+
+  def prepareRunTask(state: MvcEppBenchState): Task[Vector[Throwable]] = {
+    val par = state.term.getOrElse(throw new Error("Failed to prepare executable rholand term"))
+    state.runtime.reducer
+      .inj(par)(state.rand)
+      .map(_ => state.runtime.readAndClearErrorVector())
+  }
+
+  //if we run multiple tests on a single-threaded scheduler
+  //they will compete on single execution queue
+  //therefore this test always limited to one thread
+  @Benchmark @Threads(1)
+  def replayMvcEppST(state: MvcEppBenchState): Unit = {
+    val runTask = prepareRunTask(state).executeOn(state.singleThreadedScheduler, forceAsync = false)
+    processErrors(
+      runTask.runSyncUnsafe(Duration.Inf)(state.singleThreadedScheduler, CanBlock.permit))
+  }
+
+  @Benchmark
+  def replayMvcEppMT(state: MvcEppBenchState): Unit = {
+    implicit val scheduler: Scheduler = monix.execution.Scheduler.Implicits.global
+    val runTask                       = prepareRunTask(state)
+    processErrors(Await.result(runTask.runAsync, Duration.Inf))
+  }
+}
+
+object EvalBench {
+
+  @State(Scope.Benchmark)
+  class MvcEppBenchState extends BenchStateBase {
+    val singleThreadedScheduler: Scheduler = TrampolineScheduler.apply(
+      Scheduler.singleThread(name = "mvcepp-1"),
+      ExecutionModel.SynchronousExecution)
+
+    override val rhoScriptSource: String = "/rholang/mvcepp.rho"
+  }
+
+  trait BenchStateBase {
+    private val dbDir: Path   = Files.createTempDirectory("rchain-storage-test-")
+    private val mapSize: Long = 1024 * 1024 * 1024
+
+    val rhoScriptSource: String
+    val runtime: Runtime       = Runtime.create(dbDir, mapSize)
+    val rand: Blake2b512Random = Blake2b512Random(128)
+    var term: Option[Par]      = None
+
+    /**
+      * until we add flag 'delete_lmdb_dir_on_close' for benchmarks and unit-tests
+      * this prevents periodic out of disk space failures
+      */
+    def deleteOldStorages(): Unit =
+      dbDir.getParent.toFile.listFiles
+        .filter(dir => dir.isDirectory && (dir != dbDir))
+        .filter(_.getName.startsWith("rchain-storage-test-"))
+        .foreach(dir =>
+          try {
+            println(s"deleting... $dir")
+            dir.toPath.recursivelyDelete
+          } catch {
+            case _: Exception =>
+        })
+
+    @Setup
+    def doSetup(): Unit = {
+      deleteOldStorages()
+
+      def resourceFileReader(path: String): InputStreamReader =
```

Comment:
Could we move this to the outermost context, since it's a fairly general helper function?